pkg/envoy/cds/cluster.go

```diff
@@ -25,14 +25,30 @@ const (
 	clusterConnectTimeout = 1 * time.Second
 )
 
+type options struct {
+	noPermissive bool
+	noTLS        bool
+}
+
+// Option is type of function that edits the defaults of the options struct.
+type Option func(o options)
+
+// NoTLS is an option to disable TLS.
+func NoTLS(o options) {
+	o.noTLS = true
+}
+
+// NoPermissive is an option to not generate permissive endpoints discovery for clusters.
+func NoPermissive(o options) {
+	o.noPermissive = true
+}
+
 // getUpstreamServiceCluster returns an Envoy Cluster corresponding to the given upstream service
 // Note: ServiceIdentity must be in the format "name.namespace" [https://github.com/openservicemesh/osm/issues/3188]
-func getUpstreamServiceCluster(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService, cfg configurator.Configurator) (*xds_cluster.Cluster, error) {
-	clusterName := upstreamSvc.String()
-	marshalledUpstreamTLSContext, err := ptypes.MarshalAny(
-		envoy.GetUpstreamTLSContext(downstreamIdentity, upstreamSvc))
-	if err != nil {
-		return nil, err
+func getUpstreamServiceCluster(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService, cfg configurator.Configurator, opts ...Option) (*xds_cluster.Cluster, error) {
```

Comment:
Hey @steeling, I didn't mean to suggest using variadic param here, but rather an additional param to use mTLS.

The goal here is to have a generic way to build clusters for different use cases, ex.
```
getUpstreamServiceCluster(clusterConfig []someStruct) (*xds_cluster.Cluster, error)
```

Since this is outside the scope of your PR, I suggested to simply pass a bool to not initialize the upstream TLS context when using SNI passthrough.

Reply:
Sure, but we also need to disable permissive discovery. So there needs to be an additional bool option, at which variation Params would be more clean than tacking on multiple bool options